StaticAnalyzer/views/ios/common_rules.py

```diff
@@ -0,0 +1,137 @@
+"""
+Rule Format.
+
+1. desc - Description of the findings
+
+2. type
+   a. string
+   b. regex
+
+3. match
+   a. single_regex - if re.findall(regex1, input)
+   b .regex_and - if re.findall(regex1, input) and re.findall(regex2, input)
+   c. regex_or - if re.findall(regex1, input) or re.findall(regex2, input)
+   d. single_string - if string1 in input
+   e. string_and - if (string1 in input) and (string2 in input)
+   f. string_or - if (string1 in input) or (string2 in input)
+   g. string_and_or -  if (string1 in input) and ((string_or1 in input)
+                       or (string_or2 in input))
+   h. string_or_and - if (string1 in input) or ((string_and1 in input)
+                      and (string_and2 in input))
+
+4. level
+   a. high
+   b. warning
+   c. info
+   d. good
+
+5. input_case
+   a. upper
+   b. lower
+   c. exact
+
+6. others
+   a. string<no> - string1, string2, string3, string_or1, string_and1
+   b. regex<no> - regex1, regex2, regex3
+
+"""
+COMMON_RULES = [
+    {
+        'desc': 'IP Address disclosure',
+        'type': 'regex',
+        'regex1': r'\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}',
+        'level': 'warning',
+        'match': 'single_regex',
+        'input_case': 'exact',
+        'cvss': 4.3,
+        'cwe': 'CWE-200',
+        'owasp': '',
+    },
+    {
+        'desc': ('Files may contain hardcoded sensitive'
+                 ' informations like usernames, passwords, keys etc.'),
+        'type': 'regex',
+        'regex1': (r'(password\s*=\s*[\'|\"].+[\'|\"]\s{0,5})|'
+                   r'(pass\s*=\s*[\'|\"].+[\'|\"]\s{0,5})|'
+                   r'(username\s*=\s*[\'|\"].+[\'|\"]\s{0,5})|'
+                   r'(secret\s*=\s*[\'|\"].+[\'|\"]\s{0,5})|'
+                   r'(key\s*=\s*[\'|\"].+[\'|\"]\s{0,5})'),
+        'level': 'high',
+        'match': 'single_regex',
+        'input_case': 'lower',
+        'cvss': 7.4,
+        'cwe': 'CWE-312',
+        'owasp': 'M9: Reverse Engineering',
+    },
+    {
+        'desc': ('App uses SQLite Database. '
+                 'Sensitive Information should be encrypted.'),
+        'type': 'string',
+        'string1': 'sqlite3_exec',
```

Comment:
Is this applicable to Swift?

Reply:
Yes, but I see that sqlite3 may be used with different function than exec.
Maybe we should detect patter like `sqlite3_****(` with regex `sqlite3[_][\S]*\(`.
So I could leave the `sqlite3_exec` in objc and add this regex in Swift.

Reply:
We only need to consider the `sqlite3_**` functions that execute raw SQL query. If you know of other functions with the prefix, we should consider them. My question was more like is the function `sqlite3_exec` having same syntax/function name is Swift as well?

Reply:
Yes, sqlite3_exec is applicable in Swift, It has the same syntax.
To run sqlite statement We have to execute `sqlite3_finalize` function, I will add it with next part of the rules.